kratos/includes/accessor.h

```diff
@@ -0,0 +1,279 @@
+     * @param rVariable The variable considered (array_1d<double, 6 > type properties)
+     * @param rProperties The properties considered
+     * @param rGeometry The geometry considered
+     * @param rShapeFunctionVector The shape function of the GP considered
+     * @param rProcessInfo The process info considered
+     * @return The array_1d<double, 6 > type properties
+     */
+    virtual array_1d<double, 6> GetValue(
+        const Variable<array_1d<double, 6>>& rVariable,
+        const Properties& rProperties,
+        const GeometryType& rGeometry,
+        const Vector& rShapeFunctionVector,
+        const ProcessInfo& rProcessInfo
+        ) const;
+
+    /**
+     * @brief Custom method to retrieve array_1d<double, 6 > type properties
+     * @param rVariable The variable considered (array_1d<double, 6 > type properties)
+     * @param rProperties The properties considered
+     * @param rGeometry The geometry considered
+     * @param rShapeFunctionVector The shape function of the GP considered
+     * @param rProcessInfo The process info considered
+     * @return The array_1d<double, 6 > type properties
+     */
+    virtual array_1d<double, 4> GetValue(
+        const Variable<array_1d<double, 4>>& rVariable,
+        const Properties& rProperties,
+        const GeometryType& rGeometry,
+        const Vector& rShapeFunctionVector,
+        const ProcessInfo& rProcessInfo
+        ) const;
+
+    /**
+     * @brief Custom method to retrieve array_1d<double, 6 > type properties
+     * @param rVariable The variable considered (array_1d<double, 6 > type properties)
+     * @param rProperties The properties considered
+     * @param rGeometry The geometry considered
+     * @param rShapeFunctionVector The shape function of the GP considered
+     * @param rProcessInfo The process info considered
+     * @return The array_1d<double, 6 > type properties
+     */
+    virtual array_1d<double, 9> GetValue(
+        const Variable<array_1d<double, 9>>& rVariable,
+        const Properties& rProperties,
+        const GeometryType& rGeometry,
+        const Vector& rShapeFunctionVector,
+        const ProcessInfo& rProcessInfo
+        ) const;
+
+    /**
+     * @brief Custom method to retrieve string type properties
+     * @param rVariable The variable considered (string type properties)
+     * @param rProperties The properties considered
+     * @param rGeometry The geometry considered
+     * @param rShapeFunctionVector The shape function of the GP considered
+     * @param rProcessInfo The process info considered
+     * @return The string type properties
+     */
+    virtual std::string GetValue(
+        const Variable<std::string>& rVariable,
+        const Properties& rProperties,
+        const GeometryType& rGeometry,
+        const Vector& rShapeFunctionVector,
+        const ProcessInfo& rProcessInfo
+        ) const;
+
+    // Getting a pointer to the class
+    virtual Accessor::Pointer Clone() const;
```

Comment:
Shouldn't we add a `Create` method as well?

Reply:
We've been discussing about this in the @KratosMultiphysics/technical-committee. We'd leave this for the future, as it is not clear at all that we'll be able to standarize the accessors to be factory-based.